packages/material-ui/src/Menu/Menu.js

```diff
@@ -115,13 +176,65 @@ const Menu = React.forwardRef(function Menu(props, ref) {
     }
   });
 
+  const handleSetOpenSubMenuIndex = (value) => {
+    if (value === null) {
+      if (contentAnchorRef.current.parentElement) {
+        contentAnchorRef.current.parentElement.children[openSubMenuIndex].focus();
+      }
+    }
+    setOpenSubMenuIndex(value);
+  };
+
   const items = React.Children.map(children, (child, index) => {
+    if (!React.isValidElement(child)) {
+      return undefined;
+    }
+
+    const { subMenu, onMouseMove: onMouseMoveChildProp } = child.props;
+    const { anchorEl } = other;
+
+    const hasSubMenu = Boolean(subMenu);
+    const parentMenuOpen = Boolean(anchorEl);
```

Comment:
[`anchorEl` can also be a function](https://next--material-ui.netlify.app/api/menu/#props). Could you add a test to see what happens for both types of `anchorEl`?

Reply:
Apologies, I'm realizing I missed several of your responses that were buried in collapsed parts of the conversation and I haven't been getting emails for all of these, it seems.

I'll take a look at this once I get past the latest issues I seem to have introduced during merge conflict resolution for the rebase on the emotion changes.